lib/kubeclient/watch_stream.rb

```diff
@@ -1,54 +1,63 @@
 require 'json'
-require 'net/http'
+require 'http'
 module Kubeclient
   module Common
     # HTTP Stream used to watch changes on entities
     class WatchStream
       def initialize(uri, http_options, format: :json)
         @uri = uri
-        @http = nil
-        @http_options = http_options.merge(read_timeout: nil)
+        @http_client = nil
+        @http_options = http_options
         @format = format
       end
 
       def each
         @finished = false
-        @http = Net::HTTP.start(@uri.host, @uri.port, @http_options)
 
-        buffer = ''
-        request = generate_request
+        @http_client = build_client
+        response = @http_client.request(:get, @uri, build_client_options)
+        unless response.code < 300
```

Comment:
@msufa how is this related to "Use non-blocking IO"? It's a good improvement but it should be evaluated separately, so if it's not part of this PR please remove it.